cmd/lotus-chainwatch/sync.go

```diff
@@ -200,7 +200,7 @@ func syncHead(ctx context.Context, api api.FullNode, st *storage, ts *types.TipS
 					log.Error(err)
 					return
 				}
-				ast, err := api.StateReadState(ctx, &act, ts.Key())
+				ast, err := api.StateReadState(ctx, &act, pts.Key())
```

Comment:
I believe this was a bug, the `ts` is the tipset syncHead was called with.

Reply:
looks like a bug to me, nice catch